apps/trading/pages/markets/trade-grid.tsx

```diff
@@ -31,57 +34,110 @@ interface TradeGridProps {
   market: Market_market;
 }
 
+export const TradeMarketHeader = ({ market }: TradeGridProps) => {
+  const [open, setOpen] = useState(false);
+  const candlesClose: string[] = (market?.candles || [])
+    .map((candle) => candle?.close)
+    .filter((c): c is CandleClose => c !== null);
```

Comment:
It is easier to read, but I ended up doing this (even mapping first and filtering after) because a simple null check would not suffice on the Appolo generated types, and I needed a required type for CandleClose.  
Also, this array has a constant bound, it would be of a maximum of 24 elements, so this iteration does not seem to represent any complexity issue.